spec/policy/rate_limit/rate_limit_spec.lua

```diff
@@ -0,0 +1,175 @@
+local RateLimitPolicy = require('apicast.policy.rate_limit')
+local function init_val()
+  ngx.var = {}
+  ngx.var.request_time = '0.060'
+
+  ngx.shared.limiter = {}
+  ngx.shared.limiter.get = function(_, key)
+    return ngx.shared.limiter[key]
+  end
+  ngx.shared.limiter.set = function(_, key, val)
+    ngx.shared.limiter[key] = val
+  end
+  ngx.shared.limiter.incr = function(_, key, val, init)
+    local v = ngx.shared.limiter[key]
+    if not v then
+      ngx.shared.limiter[key] = val + init
+    else
+      ngx.shared.limiter[key] = v + val
+    end
+    return ngx.shared.limiter[key]
+  end
+  ngx.shared.limiter.expire = function(_, _, _)
+    return true, nil
+  end
+end
+
+describe('Rate limit policy', function()
+  local ngx_exit_spy
+  local ngx_sleep_spy
+
+  setup(function()
+    ngx_exit_spy = spy.on(ngx, 'exit')
+    ngx_sleep_spy = spy.on(ngx, 'sleep')
+  end)
+
+  before_each(function()
+    local redis = require('resty.redis'):new()
+    redis:connect('127.0.0.1', 6379)
+    redis:select(1)
+    redis:del('test1', 'test2', 'test3')
+    init_val()
+  end)
+
+  describe('.access', function()
+    it('success with multiple limiters', function()
+      local config = {
+        limiters = {
+          {name = "connections", key = 'test1', conn = 20, burst = 10, delay = 0.5},
+          {name = "leaky_bucket", key = 'test2', rate = 18, burst = 9},
+          {name = "fixed_window", key = 'test3', count = 10, window = 10}
+        },
+        redis_url = 'redis://localhost:6379/1'
```

Comment:
Some of the tests use `redis://localhost`. This will not work in some environments like the dockerized development one. 

No need to fix this now. This is just a note so we remember to fix it in a different PR. It would be good to use an env with the Redis host like we do in the integration tests with `TEST_NGINX_REDIS_HOST`.